### PR TITLE
docs: Rework building and image creation instructions

### PR DESCRIPTION
**Pull request by 64**
This PR changes the following:
- add instructions for installing pre-built packages
- rework image creation section

[Rendered](https://github.com/managarm/managarm/blob/e2b3bb538e846a727b349a3a1ba6f71bc90bae1c/docs/src/building/index.md)

### CODE REVIEW

**Comment by 64** [comment]
The instructions given here require the changes from `https://github.com/managarm/xbstrap/pull/50`.

File: docs/src/building/index.md [hunk 1]
```diff
@@ -62,88 +62,68 @@ source and build directories. Here we use `~/managarm`, but it can be any direct
...
+	Going with method 1 will require `libguestfs` to be installed on the host.
+	After installing `libguestfs` it might be necessary to run the following:
+	```bash
+	sudo install -d /usr/lib/guestfs
+	sudo update-libguestfs-appliance
+	```
+1.  Add the following to `build/bootstrap-site.yml`, depending on what mount method you have chosen:
+	```yaml
+	define_options:
+		mount-using: 'loopback' # or guestfs/docker
```
Comment (avdgrinten) [inline docs/src/building/index.md:112]: Broken indentation.

File: docs/src/building/index.md [hunk 2]
```diff
@@ -62,88 +62,68 @@ source and build directories. Here we use `~/managarm`, but it can be any direct
...
+	```bash
+	xbstrap run make-image
+	```
+1.  Launch the image using QEMU:
+	```bash
+	xbstrap run qemu # Note that you can combine the last two operations: xbstrap run make-image qemu
+	```
+
+Alternatively, you can call the necessary scripts manually (check their help messages for more information):
+* `../src/scripts/update-image.py` for copying the files onto the image (the mount method can be selected with the `--mount-using` argument)
```
Comment (avdgrinten) [inline docs/src/building/index.md:128]:
```suggestion
* `../src/managarm/tools/gen-initrd.py` and `../src/scripts/update-image.py` for copying the files onto the image (the mount method can be selected with the `--mount-using` argument)
```

File: docs/src/building/index.md [hunk 3]
```diff
@@ -62,88 +62,68 @@ source and build directories. Here we use `~/managarm`, but it can be any direct
...
-After managarm's packages have been built, building a HDD image of the system
-is straightforward. The [image_create.sh](https://github.com/qookei/image_create) script
-can be used to create an empty HDD image.
-
-Download the `image_create.sh` script and mark it executable:
-```bash
-wget 'https://raw.githubusercontent.com/qookei/image_create/master/image_create.sh'
-chmod +x image_create.sh
-```
+	> Note: **this only works if you use `cbuildrt`**. Docker and containerless builds must do a full build from source (see below).
```
[the remark this answers is not in the document]
Comment (64) [reply docs/src/building/index.md]: Makes sense - I think our generic 'Docker is a bit untested' warning on the Docker instructions should suffice for that (plus my change)

**Comment by avdgrinten** [comment]
LGTM now. Can you squash the fixup commits into the earlier ones?

**Comment by 64** [comment]
In the latest force push I split the 'create bootstrap-site.yml and edit rootfs key as appropriate' into two instructions.

**Comment by avdgrinten** [comment]
Now, that the blockers are gone, this can be merged. Thank you!

**Pull request merged**